apps/oc_chef_wm/itest/oc_chef_wm_keys_SUITE.erl

```diff
@@ -194,16 +336,64 @@ init_per_testcase(list_user_no_keys, Config) ->
     make_user(Config, ?ADMIN_USER_NAME, ?ADMIN_AUTHZ_ID),
     %chef_db:fetch(#chef_user{username = ?ADMIN_USER_NAME}, context()),
     Config;
+init_per_testcase(get_user_default_key,  Config) ->
+    make_user(Config, ?USER_NAME, ?USER_AUTHZ_ID),
+    Config;
+init_per_testcase(get_client_default_key, Config) ->
+    make_client(Config, ?CLIENT_NAME),
+    Config;
+init_per_testcase(get_client_multiple_keys, Config) ->
+    make_client(Config, ?CLIENT_NAME),
+    ClientId = client_id(Config, ?CLIENT_NAME),
+    add_key(Config, ClientId, ?KEY1NAME, ?KEY1EXPIRE),
+    add_key(Config, ClientId, ?KEY2NAME, ?KEY2EXPIRE),
+    Config;
+init_per_testcase(get_user_multiple_keys, Config) ->
+    make_user(Config, ?USER_NAME, ?USER_AUTHZ_ID),
+    UserId = user_id(?USER_NAME),
+    add_key(Config, UserId, ?KEY1NAME, ?KEY1EXPIRE),
+    add_key(Config, UserId, ?KEY2NAME, ?KEY2EXPIRE),
+    Config;
+init_per_testcase(get_client_no_keys, Config) ->
+    make_client(Config, ?CLIENT_NAME),
+    sqerl:adhoc_delete(<<"keys">>, all),
+    % make this user after clearing keys, so that we have a user
+    % who can make the request.
+    make_user(Config, ?ADMIN_USER_NAME, ?ADMIN_AUTHZ_ID),
+    Config;
+init_per_testcase(get_user_no_keys, Config) ->
+    make_user(Config, ?USER_NAME, ?USER_AUTHZ_ID),
+    sqerl:adhoc_delete(<<"keys">>, all),
+    make_user(Config, ?ADMIN_USER_NAME, ?ADMIN_AUTHZ_ID),
+    Config;
+init_per_testcase(get_client_wrong_key, Config) ->
+    make_user(Config, ?ADMIN_USER_NAME, ?ADMIN_AUTHZ_ID),
+    make_client(Config, ?CLIENT_NAME),
+    Config;
+init_per_testcase(get_user_wrong_key, Config) ->
+    make_user(Config, ?ADMIN_USER_NAME, ?ADMIN_AUTHZ_ID),
+    make_user(Config, ?USER_NAME, ?USER_AUTHZ_ID),
+    Config;
+init_per_testcase(get_key_for_nonexistant_user, Config) ->
+    make_user(Config, ?ADMIN_USER_NAME, ?ADMIN_AUTHZ_ID),
+    Config;
+init_per_testcase(get_key_for_nonexistant_client, Config) ->
+    make_user(Config, ?ADMIN_USER_NAME, ?ADMIN_AUTHZ_ID),
+    Config;
 init_per_testcase(post_client_new_valid_key, Config) ->
     make_admin_and_client(Config);
+init_per_testcase(post_new_key_invalid_date, Config) ->
```

Comment:
Might be able to use a guard? 
    init_per_testcase(TestCase, Config) -> when TestCase =:= post_new_key_invalid_date; 
                                                                            TestCase =:= post_new_key_invalid_digits_date
                                                                            TestCase =:= post_new_key_invalid_utc_date
etc?